src/base/static/components/place-detail-view/place-detail-survey.js

```diff
@@ -0,0 +1,192 @@
+// TODO: localization in this component.
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import {
+  Map as ImmutableMap,
+  OrderedMap as ImmutableOrderedMap,
+} from "immutable";
+import emitter from "../utils/emitter";
+
+import FormField from "../form-field";
+import SecondaryButton from "../ui-elements/secondary-button";
+import PlaceDetailSurveyResponse from "./place-detail-survey-response";
+import WarningMessagesContainer from "../ui-elements/warning-messages-container";
+import Avatar from "../ui-elements/avatar";
+
+import constants from "../constants";
+import { placeDetailSurvey as messages } from "../messages";
+
+import "./place-detail-survey.scss";
+
+class PlaceDetailSurvey extends Component {
+  constructor(props) {
+    super(props);
+    let fields = ImmutableOrderedMap();
+    this.props.surveyConfig.items.forEach(field => {
+      fields = fields.set(
+        field.name,
+        ImmutableMap()
+          .set(constants.FIELD_STATE_RENDER_KEY, Math.random())
+          .set(constants.FIELD_STATE_VALUE_KEY, "")
+      );
+    });
+    this.state = {
+      fields: fields,
+      isFormSubmitting: false,
+      formValidationErrors: new Set(),
+      showValidityStatus: false,
+    };
+  }
+
+  componentWillMount() {
+    emitter.addListener("place-detail-survey:save", () => {
+      let fields = ImmutableOrderedMap();
+      this.props.surveyConfig.items.forEach(field => {
+        fields = fields.set(
+          field.name,
+          ImmutableMap()
+            .set(constants.FIELD_STATE_RENDER_KEY, Math.random())
+            .set(constants.FIELD_STATE_VALUE_KEY, "")
+        );
+      });
```

Comment:
This code is exactly the same as above. Maybe it should be in an "initializeFields" function?